### ISSUE TITLE
Implement the variance metric

### ISSUE THREAD

**Issue by sarambulo** [nikitachaudharicodes/ProjeXion#5]
Create a `variance.py` module inside the `models` directory that implements the class `VarianceLayer`. This class should inherit from `torch.nn.Module` and should take as input a NxHxWxDx32 tensors and generate an output tensor of size HxWxDx32 by calculating the element wise variance over the first dimension (dim 0).

![Image](https://github.com/user-attachments/assets/574a1b21-8523-42b3-b610-205482aa653a)
